### PR TITLE
[Doc][Improve] improve pom

### PR DESCRIPTION
**Pull request by EricJoy2048**
1.Improve seatunnel-dist/pom.xml and will package all jdbc driver and hadoop jars in install package.
2.Put `hazelcast.yaml` `hazelcast-client.yaml` `seatunnel.yaml` in `config` dir.

### CODE REVIEW

File: config/hazelcast-client.yaml [hunk 1]
```diff
@@ -15,6 +15,10 @@
...
-    engine:
-        backup-count: 1
\ No newline at end of file
+hazelcast-client:
+  cluster-name: seatunnel
+  properties:
+      hazelcast.logging.type: log4j2
+  network:
+    cluster-members:
+      - localhost:5801
```
Comment (CalvinKirs) [inline config/hazelcast-client.yaml:24]: We also need detailed documentation for various parameters

File: config/log4j2.properties [hunk 2]
```diff
@@ -16,63 +16,32 @@
 # limitations under the License.
 ################################################################################
 
-property.file_path = ${sys:seatunnel.logs.path:-/tmp/seatunnel/logs}
-property.file_name = ${sys:seatunnel.logs.file_name:-seatunnel}
-property.file_split_size = 100MB
-property.file_count = 100
-property.file_ttl = 7d
+appender.console.type=Console
```
Comment (hailin0) [inline config/log4j2.properties]: Why change this file?

Comment (EricJoy2048) [reply config/log4j2.properties]: Because I found the file log can not work in seatunnel cluster.

File: config/hazelcast-client.yaml (hunk 1 above)
Comment (EricJoy2048) [reply config/hazelcast-client.yaml:24]: yes.

File: seatunnel-engine/seatunnel-engine-common/src/main/resources/hazelcast.yaml [hunk 3]
```diff
@@ -32,7 +27,11 @@ hazelcast:
...
-    hazelcast.tcp.join.port.try.count: 30
-    hazelcast.logging.type: log4j2
+  map:
+    map-name-template:
+      map-store:
+        enabled: true
+        initial-mode: EAGER
+        class-name: org.apache.seatunnel.engine.server.persistence.FileMapStore
+        properties:
+          path: /tmp/file-store-map
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-common/src/main/resources/hazelcast.yaml]: We only need to persist part of the IMap.
It should be like this.
```suggestion
  map:
    engine*:
      map-store:
        enabled: true
        initial-mode: EAGER
        factory-class-name: org.apache.seatunnel.engine.server.persistence.FileMapStoreFactory
        properties:
          namespace: /tmp/seatunnel/imap
          clusterName: seatunnel-clsuter
          fs.defaultFS: file:///
```

Comment (liugddx) [reply seatunnel-engine/seatunnel-engine-common/src/main/resources/hazelcast.yaml]: The IMap we need to persist should have a uniform prefix, like `engine_runningJobInfo`

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/resources/hazelcast.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      port-count: 100
+      port: 5801
+  map:
+    map-name-template:
+      map-store:
+        enabled: true
+        initial-mode: EAGER
+        class-name: org.apache.seatunnel.engine.server.persistence.FileMapStore
+        properties:
+          path: /tmp/file-store-map
```
Comment (liugddx) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/resources/hazelcast.yaml]:
```suggestion
  map:
    engine*:
      map-store:
        enabled: true
        initial-mode: EAGER
        factory-class-name: org.apache.seatunnel.engine.server.persistence.FileMapStoreFactory
        properties:
          namespace: /tmp/seatunnel/imap
          clusterName: seatunnel-clsuter
          fs.defaultFS: file:///
```

File: seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/pom.xml [hunk 5]
```diff
@@ -37,9 +45,16 @@
...
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-java-sdk-bundle</artifactId>
+            <version>1.11.199</version>
+            <scope>provided</scope>
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/org.apache.flink/flink-shaded-hadoop-3-uber -->
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-shaded-hadoop-3-uber</artifactId>
+            <version>3.1.1.7.2.9.0-173-9.0</version>
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/pom.xml:57]: Manage versions in the parent pom.

File: seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/pom.xml [hunk 6]
```diff
@@ -37,9 +45,16 @@
             <scope>provided</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-client</artifactId>
-            <version>3.0.0</version>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-java-sdk-bundle</artifactId>
+            <version>1.11.199</version>
```
Comment (liugddx) [inline seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/pom.xml:50]: Manage versions in the parent pom.

File: config/seatunnel.yaml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+            timeout: 10000
+            max-concurrent: 5
+            tolerable-failure: 2
+            storage:
+                type: hdfs
+                max-retained: 3
+                storageNameSpace: /tmp/seatunnel/checkpoint_snapshot
+                plugin-config:
+                    storage-type: hdfs
+                        fs.defaultFS: /tmp/ # Ensure that the directory has written permission
```
Comment (liugddx) [inline config/seatunnel.yaml]:
```suggestion
                plugin-config:
                    storage-type: hdfs
                    fs.defaultFS: /tmp/ # Ensure that the directory has written permission
```

File: seatunnel-engine/seatunnel-engine-common/src/main/resources/hazelcast.yaml (hunk 3 above)
Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-common/src/main/resources/hazelcast.yaml]: > The IMap we need to persist should have a uniform prefix, like `engine_runningJobInfo`

Thanks, I will fix it.

File: seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/pom.xml (hunk 6 above)
Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/pom.xml:50]: done

File: seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/pom.xml (hunk 5 above)
Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-storage/checkpoint-storage-plugins/checkpoint-storage-hdfs/pom.xml:57]: done

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/resources/hazelcast.yaml (hunk 4 above)
Comment (EricJoy2048) [reply seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/resources/hazelcast.yaml]: done

File: seatunnel-engine/seatunnel-engine-common/src/main/resources/hazelcast.yaml (hunk 3 above)
Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-common/src/main/resources/hazelcast.yaml]: done

Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-common/src/main/resources/hazelcast.yaml]: > The IMap we need to persist should have a uniform prefix, like `engine_runningJobInfo`

@ic4y  PTAL

File: config/seatunnel.yaml (hunk 7 above)
Comment (EricJoy2048) [reply config/seatunnel.yaml]: done

File: seatunnel-engine/seatunnel-engine-common/src/main/resources/hazelcast.yaml (hunk 3 above)
Comment (ic4y) [reply seatunnel-engine/seatunnel-engine-common/src/main/resources/hazelcast.yaml]: The FileMapStoreFactory has not been merged into #3637, so there is no need to add map-store configuration at this pr. @liugddx

File: config/log4j2.properties (hunk 2 above)
Comment (hailin0) [reply config/log4j2.properties]: If you need to output logs to files, you need to enable `fileAppender` and close `console**Appender` 

example

```properties
############################ log output to console #############################
#rootLogger.appenderRef.consoleStdout.ref = consoleStdoutAppender
#rootLogger.appenderRef.consoleStderr.ref = consoleStderrAppender
############################ log output to console #############################
############################ log output to file    #############################
rootLogger.appenderRef.file.ref = fileAppender
############################ log output to file    #############################
```

**Comment by TyrantLucifer** [comment]
Please fix CI.

File: docs/en/seatunnel-engine/cluster-mode.md [hunk 8]
```diff
@@ -1,6 +1,24 @@
...
 ---
 
 # Run Job With Cluster Mode
 
+This is the most recommended way to use SeaTunnel Engine in the production environment. Full functionality of SeaTunnel Engine is supported in this mode and the cluster mode will have better performance and stability. 
+
+In the cluster mode, the SeaTunnel Engine cluster needs to be deployed first, and the client will submit the job to the SeaTunnel Engine cluster for running.
+
+## Deploy SeaTunnel Engine Cluster
+Deploy a SeaTunnel Engine Cluster reference [SeaTunnel Engine Cluster Deploy](deployment.md)
```
Comment (TaoZex) [inline docs/en/seatunnel-engine/cluster-mode.md:12]: Put a blank line in the middle

File: docs/en/seatunnel-engine/deployment.md [hunk 9]
```diff
@@ -1,8 +1,189 @@
...
+
+Like Flink, SeaTunnel Engine support Chandy–Lamport algorithm. Therefore, SeaTunnel Engine can realize data synchronization without data loss and duplication.
+
+**interval**
+
+The interval between two checkpoints, unit is milliseconds. If the `checkpoint.interval` parameter is configured in the `env` of the job config file, the value set here will be overwritten.
+
+**timeout**
+
+The timeout of a checkpoint. If a checkpoint cannot be completed within the timeout period, a checkpoint failure will be triggered. Job will be restored.
```
Comment (TaoZex) [inline docs/en/seatunnel-engine/deployment.md:75]:
```suggestion
The timeout of a checkpoint. If a checkpoint cannot be completed within the timeout period, a checkpoint failure will be triggered. Therefore, Job will be restored.
```

File: docs/en/seatunnel-engine/deployment.md [hunk 10]
```diff
@@ -1,8 +1,189 @@
...
+
+```
+
+TCP is our suggest way in a standalone SeaTunnel Engine cluster. 
+
+On the other hand, Hazelcast provides some other service discovery methods. For details, please refer to [hazelcast network](https://docs.hazelcast.com/imdg/4.1/clusters/setting-up-clusters)
+
+## 6. Config SeaTunnel Engine Client
+
+All SeaTunnel Engine Client config in `hazelcast-client.yaml`
```
Comment (TaoZex) [inline docs/en/seatunnel-engine/deployment.md:151]:
```suggestion
All SeaTunnel Engine Client config in `hazelcast-client.yaml`.
```

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java [hunk 11]
```diff
@@ -101,7 +103,7 @@ public void init(HadoopConf conf, String jobId, String uuidPrefix, int subTaskIn
         this.jobId = jobId;
         this.subTaskIndex = subTaskIndex;
         this.uuidPrefix = uuidPrefix;
-        FileSystemUtils.CONF = getConfiguration(hadoopConf);
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/AbstractWriteStrategy.java:104]: Please revert these files.

**Comment by EricJoy2048** [comment]
The pr been split to https://github.com/apache/incubator-seatunnel/pull/3768, https://github.com/apache/incubator-seatunnel/pull/3756, https://github.com/apache/incubator-seatunnel/pull/3755, https://github.com/apache/incubator-seatunnel/pull/3751.

So, I will close this pr.

Thanks for everyone.

**Pull request closed without merging**